### DISCUSSION TITLE
Feature Request: Add option to preserve pre 0.8.0 arrow string types

### DISCUSSION THREAD

**Discussion by cpcloud** [duckdb/duckdb#7591] [Ideas]
It seems that duckdb 0.8.0 now returns string columns with the arrow type `large_string` when exporting to arrow.

Would it be possible to add a setting to disable this so that the previous behavior is still available?

`large_string` breaks some tests in the ibis test suite and we'd like to avoid returning different types from previous versions without the user explicitly opting in.

**Comment by pdet** [comment]
I did already because of ArrowJS https://github.com/duckdb/duckdb/pull/7540
I can expose that in the Python client if that helps

**Reply by dude0001** [reply]
@pdet I'm running into this issue using `fetch_record_batch` in the Python client and sending Arrow IPC to JavaScript. Is there a way to disable the use of `large_string` in the Python Client? I see the `offset_size` option added in Arrow Options, but it does not seem to be available in Python.

**Reply by Tishj** [reply]
We should probably add a pragma/setting to set the default arrow buffer size in the connection, then that can be used in places that don't have the option?

Or we should just add the option to `fetch_record_batch` as a keyword parameter, but that seems like a local fix, there are probably other things that are also missing the option.

**Comment by cpcloud** [comment]
Seems like this is no longer an issue in 0.8.1:

```
In [8]: import duckdb

In [9]: duckdb.__version__
Out[9]: '0.8.1'

In [10]: duckdb.connect().execute("select 'some text' as a").fetch_arrow_table()
Out[10]:
pyarrow.Table
a: string
----
a: [["some text"]]
```

**Reply by Mause** [reply]
Yep, this was fixed in https://github.com/duckdb/duckdb/pull/7784

**Reply by cpcloud** [reply]
It might be useful to leave the discussion open and mark my answer as accepted. I believe this may help search engines surface it more easily.

**Reply by Mause** [reply]
I don't have access to mark an answer as accepted, I think you might though as the asker?
